zallet/src/components/json_rpc/methods/view_transaction.rs

```diff
@@ -472,9 +676,141 @@ pub(crate) fn call(wallet: &DbConnection, txid_str: &str) -> Response {
+        // rejected by the backing full node.
+        .map_err(|e| LegacyCode::Database.with_message(e.to_string()))?;
+
     Ok(Transaction {
-        txid: txid_str.into(),
+        txid: txid_str.to_ascii_lowercase(),
+        status: wallet_tx_info.status,
+        confirmations: wallet_tx_info.confirmations,
+        blockhash: wallet_tx_info.blockhash,
+        blockindex: wallet_tx_info.blockindex,
+        blocktime: wallet_tx_info.blocktime,
+        version: tx.version().header() & 0x7FFFFFFF,
+        expiryheight: wallet_tx_info.expiryheight,
+        fee: fee.map(value_from_zatoshis),
+        generated: wallet_tx_info.generated,
         spends,
         outputs,
     })
 }
+
+struct WalletTxInfo {
+    status: &'static str,
+    confirmations: i32,
+    generated: Option<bool>,
+    blockhash: Option<String>,
+    blockindex: Option<u16>,
+    blocktime: Option<u64>,
+    expiryheight: u64,
+}
+
+impl WalletTxInfo {
+    /// Logic adapted from `WalletTxToJSON` in `zcashd`, to match the semantics of the `gettransaction` fields.
+    async fn fetch(
+        wallet: &DbConnection,
+        chain: &FetchServiceSubscriber,
+        tx: &zcash_primitives::transaction::Transaction,
+        chain_height: BlockHeight,
+    ) -> Result<Self, SqliteClientError> {
+        let mined_height = wallet.get_tx_height(tx.txid())?;
+
+        let confirmations = {
+            match mined_height {
+                Some(mined_height) => (chain_height + 1 - mined_height) as i32,
+                None => {
+                    // TODO: Also check if the transaction is in the mempool for this branch.
+                    -1
+                }
+            }
+        };
+
+        let generated = if tx
+            .transparent_bundle()
+            .is_some_and(|bundle| bundle.is_coinbase())
+        {
+            Some(true)
+        } else {
+            None
+        };
+
+        let mut status = "waiting";
+
+        let (blockhash, blockindex, blocktime) = if let Some(height) = mined_height {
+            status = "mined";
+
+            // TODO: Once Zaino updates its API to support atomic queries, it should not
+            // be possible to fail to fetch the block that a transaction was observed
+            // mined in.
+            let block_metadata = wallet
+                .block_metadata(height)?
+                // This would be a race condition between this and a reorg.
+                .ok_or(SqliteClientError::ChainHeightUnknown)?;
+            let block = chain
+                .get_block(BlockId {
+                    height: 0,
+                    hash: block_metadata.block_hash().0.to_vec(),
+                })
+                .await
+                .map_err(|_| SqliteClientError::ChainHeightUnknown)?;
+
+            let tx_index = block
+                .vtx
+                .iter()
+                .find(|tx| tx.hash == block_metadata.block_hash().0)
```

Comment:
Bug: comparing the transaction id to the block hash?

Reply:
Good catch; this is a bug in #36 that got copied over here.